src/driver.ml

```diff
@@ -65,21 +65,27 @@ end
 module Instrument = struct
   type pos = Before | After
 
-  type t = { transformation :  Parsetree.structure -> Parsetree.structure ; position : pos}
+  type t = { transformation : Expansion_context.Base.t -> Parsetree.structure -> Parsetree.structure ; position : pos}
 
-  let make transformation ~position = { transformation ; position }
+  module V2 = struct
+    let make transformation ~position = { transformation ; position }
+  end
+
+  let make transformation ~position =
+    let transformation _ st = transformation st in
+    V2.make transformation ~position
 end
 
 module Transform = struct
   type t =
     { name            : string
     ; aliases         : string list
-    ; impl            : (Parsetree.structure -> Parsetree.structure) option
-    ; intf            : (Parsetree.signature -> Parsetree.signature) option
+    ; impl            : (Expansion_context.Base.t -> Parsetree.structure -> Parsetree.structure) option
```

Comment:
How about making it a labelled argument?

It's not super flexible as I think users will still have to define their function arguments in the right order but I think it's still a bit clearer this way.

Reply:
The same apply for Instrument I guess!

Reply:
I just checked and `Deriving` and `Extension` both use a labelled arg so it would also be more consistent!

Reply:
The problem is that `Ast_traverse.map_with_expansion_context` takes an unlabelled context argument. So there already is an inconsistency between `Deriving` and `Extension` on one hand and `Ast_traverse` on the other hand. I've chosen to be consistent with `Ast_traverse`, since that's what needs to be used along with the new `Driver.V2` API (see the tests for example). Do you think it might be worth making a V2 for `Ast_traverse.map_with_expansion_context` in order to get rid of that inconsistency and make it labelled eveywhere?

Reply:
@ceastlund : I've talked about this with @NathanReb and we were wondering of you could also give us your opinion on this. 

To sum up: The question is if the expansion context should be a labelled or an unlabelled argument.
- in favor of unlabelled: in the class `Ast_traverse.map_with_expansion_context`, it is unlabelled. A common way to use the new API is to create an object `o`, that inherits from that class, and using `o#structure` and/or `o#signature` directly to register the transformation (as done in [this test](https://github.com/ocaml-ppx/ppxlib/pull/202/files#diff-664f09497a98806f341fa6cbe5d2b65091e087fbf4bdf980fb1533f8f17df639R24)). Therefore, we'd need to be consistent with that class.
- in favor of labelled: The callbacks `V2`/`V3` in both `Deriving` and `Extension` use labelled arguments. Furthermore, using a labelled argument seems more intuitive.

I'm leaning towards unlabelled, since I think not being able to use `map_with_expansion_context#structure`/`signature` right away might make the user think that that wasn't meant to be used in combination. But I'd also be happy to make it labelled. What do you think?